src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java

```diff
@@ -0,0 +1,51 @@
+package com.currencybaskets.tasks;
+
+import com.currencybaskets.dao.repository.RateRepository;
+import com.currencybaskets.dto.fixer.CurrencyRate;
+import com.currencybaskets.dto.fixer.LatestRate;
+import com.currencybaskets.exceptions.ServiceUnavailableException;
+import com.currencybaskets.services.AccountService;
+import com.currencybaskets.services.CurrencyFixerClient;
+import com.currencybaskets.view.RateUpdate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class UpdateLatestRateScheduledTask {
+
+	private static final Logger LOG = LoggerFactory.getLogger(UpdateLatestRateScheduledTask.class);
+
+	@Autowired
+	private CurrencyFixerClient fixer;
+
+	@Autowired
+	private AccountService accountService;
+
+	@Autowired
+	private RateRepository rateRepository;
+
+	//Once per day
+	@Scheduled(fixedRate = 24 * 60 * 60 * 60 * 1000L)
+	public void updateLatestRate() {
+		CurrencyRate currencyRate = fixer.getLatestRate();
+		if (!currencyRate.isSuccess()) {
+			throw new ServiceUnavailableException("Fixer service is not available");
+		}
+		LOG.info("Currency rate will be updated");
+		LatestRate rates = currencyRate.getLatestRate();
+		Long previousRateEurId = rateRepository.findLatestRateIdByCurrencyName("EUR");
+		Long previousRateUsdId = rateRepository.findLatestRateIdByCurrencyName("USD");
+		List<RateUpdate> updates = Stream.of(new RateUpdate(previousRateEurId, rates.getEurInRub()),
```

Comment:
Please log id of rate that will be updated.
Create list with Collections.asList(...)